pkg/apis/policy/v1alpha1/types_certificaterequestpolicy.go

```diff
@@ -172,118 +174,123 @@ type CertificateRequestPolicyAllowedX509Subject struct {
+	//
+	// NOTE: If `Required` is `true`, this field cannot be omitted. Setting the
+	// field to `""` instead, results in an always-false expression.
 	// +optional
 	Value *string `json:"value,omitempty"`
 
-	// Required marks this field as being a required value on the request.
-	// May only be set to true if Value is also defined.
+	// Required marks that the related field must be provided and not be an
+	// empty string.
+	// Defaults to `false`.
 	// +optional
 	Required *bool `json:"required,omitempty"`
 }
 
-// CertificateRequestPolicyConstraints define fields that, if defined, _must_
-// be satisfied by the CertificateRequest for the request to be permissible by
-// this policy. Fields that are omitted or have a value of `nil` will be
-// satisfied by any value on the corresponding attribute on the request.
+// CertificateRequestPolicyConstraints define fields that _must_ be satisfied
+// by the CertificateRequest for the request to be allowed by this policy.
+// Omitted fields will be satisfied by any value on the corresponding attribute
+// on the request.
 type CertificateRequestPolicyConstraints struct {
 	// MinDuration defines the minimum duration a certificate may be requested
 	// for.
-	// Values are inclusive (i.e. a min value of `1h` will accept a duration of
+	// Values are inclusive (i.e. a value of `1h` will accept a duration of
 	// `1h`). MinDuration and MaxDuration may be the same value.
-	// An omitted field or value of `nil` permits any minimum duration.
-	// If MinDuration is defined, a duration _must_ be requested on the
-	// CertificateRequest.
+	// If set, a duration _must_ be requested on the CertificateRequest.
+	// An omitted field implies no minimum constraint on duration.
 	// +optional
 	MinDuration *metav1.Duration `json:"minDuration,omitempty"`
 
 	// MaxDuration defines the maximum duration a certificate may be requested
 	// for.
-	// Values are inclusive (i.e. a max value of `1h` will accept a duration of
-	// `1h`). MaxDuration and MinDuration may be the same value.
-	// An omitted field or value of `nil` permits any maximum duration.
-	// If MaxDuration is defined, a duration _must_ be requested on the
-	// CertificateRequest.
+	// Values are inclusive (i.e. a value of `1h` will accept a duration of
+	// `1h`). MinDuration and MaxDuration may be the same value.
+	// If set, a duration _must_ be requested on the CertificateRequest.
+	// An omitted field implies no maximum constraint on duration.
 	// +optional
 	MaxDuration *metav1.Duration `json:"maxDuration,omitempty"`
 
-	// PrivateKey defines the shape of permissible private keys that may be used
-	// for the request with this policy.
-	// An omitted field or value of `nil` permits the use of any private key by
-	// the requestor.
+	// PrivateKey defines constraints on what shape of private key a
+	// CertificateRequest is allowed to use for its request.
+	// An omitted field implies no private key shape constraints.
```

Comment:
Again, not a fan of `implies`

Reply:
No "implies" left now - except in the file heading.